config/src/test/scala/bloop/config/ConfigCodecsSpec.scala

```diff
@@ -1,18 +1,27 @@
 package bloop.config
 
 import bloop.config.Config.File
-import org.junit.{Assert, Test}
-import java.nio.file.Paths
-import java.nio.file.Files
+import org.junit.{AfterClass, Assert, Test}
 import java.nio.charset.StandardCharsets
-import java.io.BufferedReader
-import java.io.InputStreamReader
-import scala.io.Source
+
+object ConfigCodecsSpec {
```

Comment:
When you define an object with the same name as a class, does the execution of the object take precedence over the class one? Or do both run?

Reply:
I'm pretty sure the object would run first. It's definitely the case in JUnit, where it may need to run `@BeforeClass` methods on the object before instantiating the class.

Reply:
Hmmm, but why do we need to define an object instead of a class? Can’t we have the after all method in the class? My expectation is that junit will run it fine

Reply:
JUnit's `@BeforeClass` and `@AfterClass` must be *static* methods. JUnit will look for static methods annotated with those. In Scala you can't directly write static methods, but methods in a companion object that do not clash with instance methods in the class will receive static forwarders, with the same annotations, in the class. So the only correct way to specify `@BeforeClass` and `@AfterClass` is indeed to define them in the companion object of your test class.

That doesn't make the object a direct target test for JUnit. JUnit is still only concerned with the `class`. But it will find the static forwarders in the class, and handle appropriately.

Reply:
Yeah... I know how static methods are encoded in Scala 😀 I was missing that they have to be static methods at all. Looks like this is fixed in junit 5